src/main.rs

```diff
@@ -71,21 +74,25 @@ struct MacheteArgs {
     paths: Vec<PathBuf>,
 }
 
-fn collect_paths(path: &Path, skip_target_dir: bool) -> Result<Vec<PathBuf>, walkdir::Error> {
+fn collect_paths(
+    path: &Path,
+    skip_target_dir: bool,
+    respect_ignore_files: bool,
```

Comment:
Since we're passing two bool arguments now, it'd be easy to confuse one with another. Could we pass a small struct `CollectPathSettings { skip_target_dir: bool, respect_ignore_files: bool }` instead? So the arguments are """named""".

Reply:
Done.